lamindb/_curate.py

```diff
@@ -1058,6 +1062,490 @@ def save_artifact(
         return self._artifact
 
 
+def _maybe_validation_error(nonval_keys: list[str], name: str):
+    if (n := len(nonval_keys)) > 0:
+        s = "s" if n > 1 else ""
+        are = "are" if n > 1 else "is"
+        raise ValidationError(
+            f"the following {n} key{s} passed to {name} {are} not allowed: {colors.yellow(_print_values(nonval_keys))}"
+        )
+
+
+class SOMACurator(BaseCurator):
+    """Curation flow for ``tiledbsoma``.
+
+    See also :class:`~lamindb.Curator`.
+
+    Args:
+        experiment_uri: The uri of a `tiledbsoma.Experiment`.
+        var_index: The registry fields for mapping the ``.var`` indices for measurements.
+            Should be in the form `{"measurement name": ("var column", field)}`.
+            These keys should be used in the flattened form ('{measurement name}__{column name in .var}')
+            in `.standardize` or `.add_new_from`, see the output of `.var_index`.
+        categoricals: A dictionary mapping ``.obs`` columns to a registry field.
+        obs_columns: The registry field for mapping the ``.obs`` columns.
+        using_key: A reference LaminDB instance.
```

Comment:
let's please not expose `using_key` for any new Curator class. It should disappear from all.

(Unless you tell me we absolutely need this @Zethson @sunnyosun, I continue to think that this is a horribly complicated design choice and a very bad name for an argument. I'll make a new GitHub issue on this topic or try to find where we previously discussed this).

Reply:
Agreed, let's hide it.

Reply:
I think it's useful to extend the CxG curator. By default, it uses the `laminlabs/cellxgene` instance but when extending it (e.g. with perturbations) you want to now not curate against `laminlabs/cellxgene` but your own instance.